### DISCUSSION TITLE
AttributeError: 'dict' object has no attribute 'atomic_numbers'

### DISCUSSION THREAD

**Discussion by phys-chem** [ACEsuit/mace#183] [Q&A] [answered]
When I run eval_configs.py, I meet this error:
File "/nfs/software/mace/scripts/eval_configs.py", line 151, in <module>
    main()
  File "/nfs/software/mace/scripts/eval_configs.py", line 88, in main
    z_table = utils.AtomicNumberTable([int(z) for z in model.atomic_numbers])

Could you help me? Thank you very much!

**Comment by ilyes319** [comment]
Could you please share the full error log?

**Comment by phys-chem** [comment]
> Could you please share the full error log?
Previously, I use the checkpoint file during training. Now I have finished the training, but also meet these errors. Thank you! Here is the error log:

ScaleShiftMACE(
  (node_embedding): LinearNodeEmbeddingBlock(
    (linear): Linear(2x0e -> 64x0e | 128 weights)
  )
  (radial_embedding): RadialEmbeddingBlock(
    (bessel_fn): BesselBasis(r_max=6.0, num_basis=8, trainable=False)
    (cutoff_fn): PolynomialCutoff(p=5.0, r_max=6.0)
  )
  (spherical_harmonics): SphericalHarmonics()
  (atomic_energies_fn): AtomicEnergiesBlock(energies=[-187.5987, -93.7994])
  (interactions): ModuleList(
    (0): RealAgnosticInteractionBlock(
      (linear_up): Linear(64x0e -> 64x0e | 4096 weights)
      (conv_tp): TensorProduct(64x0e x 1x0e+1x1o+1x2e+1x3o -> 64x0e+64x1o+64x2e+64x3o | 256 paths | 256 weights)
      (conv_tp_weights): FullyConnectedNet[8, 64, 64, 64, 256]
      (linear): Linear(64x0e+64x1o+64x2e+64x3o -> 64x0e+64x1o+64x2e+64x3o | 16384 weights)
      (skip_tp): FullyConnectedTensorProduct(64x0e+64x1o+64x2e+64x3o x 2x0e -> 64x0e+64x1o+64x2e+64x3o | 32768 paths | 32768 weights)
      (reshape): reshape_irreps()
    )
    (1): RealAgnosticResidualInteractionBlock(
      (linear_up): Linear(64x0e -> 64x0e | 4096 weights)
      (conv_tp): TensorProduct(64x0e x 1x0e+1x1o+1x2e+1x3o -> 64x0e+64x1o+64x2e+64x3o | 256 paths | 256 weights)
      (conv_tp_weights): FullyConnectedNet[8, 64, 64, 64, 256]
      (linear): Linear(64x0e+64x1o+64x2e+64x3o -> 64x0e+64x1o+64x2e+64x3o | 16384 weights)
      (skip_tp): FullyConnectedTensorProduct(64x0e x 2x0e -> 64x0e | 8192 paths | 8192 weights)
      (reshape): reshape_irreps()
    )
  )
  (products): ModuleList(
    (0-1): 2 x EquivariantProductBasisBlock(
      (symmetric_contractions): SymmetricContraction(
        (contractions): ModuleList(
          (0): Contraction(
            (contractions_weighting): ModuleList(
              (0-1): 2 x GraphModule()
            )
            (contractions_features): ModuleList(
              (0-1): 2 x GraphModule()
            )
            (weights): ParameterList(
                (0): Parameter containing: [torch.float32 of size 2x4x64]
                (1): Parameter containing: [torch.float32 of size 2x1x64]
            )
            (graph_opt_main): GraphModule()
          )
        )
      )
      (linear): Linear(64x0e -> 64x0e | 4096 weights)
    )
  )
  (readouts): ModuleList(
    (0): LinearReadoutBlock(
      (linear): Linear(64x0e -> 1x0e | 64 weights)
    )
    (1): NonLinearReadoutBlock(
      (linear_1): Linear(64x0e -> 16x0e | 1024 weights)
      (non_linearity): Activation [x] (16x0e -> 16x0e)
      (linear_2): Linear(16x0e -> 1x0e | 16 weights)
    )
  )
  (scale_shift): ScaleShiftBlock(scale=2.251151, shift=0.000000)
)
Traceback (most recent call last):
  File "/software/mace/scripts/eval_configs.py", line 151, in <module>
    main()
  File "/software/mace/scripts/eval_configs.py", line 108, in main
    output = model(batch.to_dict(), compute_stress=args.compute_stress)
  File "/software/mace-venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/software/mace-venv/lib/python3.10/site-packages/mace/modules/models.py", line 295, in forward
    node_e0 = self.atomic_energies_fn(data["node_attrs"])
  File "/software/mace-venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/software/mace-venv/lib/python3.10/site-packages/mace/modules/blocks.py", line 137, in forward
    return torch.matmul(x, self.atomic_energies)
RuntimeError: expected scalar type Double but found Float

**Comment by ilyes319** [comment, marked as answer]
When you run the eval, you should select `--default_dtype="float32"`.
Can you share your input to the eval script?

**Comment by phys-chem** [comment]
> When you run the eval, you should select `--default_dtype="float64"`. Can you share your input to the eval script?

I set --default_dtype="float32" in my "config.yaml". Thank you very much! I have solved it by adding '--default_dtype="float32"' in running eval_configs.py.
